src/wasm/wasm-binary.cpp

```diff
@@ -266,6 +293,7 @@ void WasmBinaryWriter::writeFunctions() {
   auto start = startSection(BinaryConsts::Section::Code);
   o << U32LEB(importInfo->getNumDefinedFunctions());
   ModuleUtils::iterDefinedFunctions(*wasm, [&](Function* func) {
+    assert(binaryLocationTrackedExpressionsForFunc.empty());
```

Comment:
It seems really weird that we are tracking per-func state on the module object and not the function object? Is that what this is doing or am I misunderstanding?

Reply:
This is temporary data used just during the writing process. We need to track the function's written expressions so that we can update just them, after the function is adjusted back due to shrinking the size LEB. So this can't be on the function object itself, as it would live forever there.)

Reply:
Oh, i am wrong, it's on the binary writer object, not the module object.